wordpress-mobile/gutenberg-mobile
# Split react native android artifact publish step in Buildkite

This PR splits the `react-native-aztec` Android artifact publishing to a separate Buildkite step. The main benefit of doing this is to allow retrying `react-native-bridge` step in isolation.

In the previous flow (including the one we used in CircleCI) if `react-native-bridge` failed due to temporary error such as Jitpack or jcenter not serving a dependency, we couldn't restart the build because `react-native-aztec` would have already been published. If the build was retried, trying to publish `react-native-aztec` would fail with "already published" error, so we'd require a new commit and restart all tasks. (including CircleCI ones)

In this new version, we publish `react-native-aztec` while we are bundling the Android JS file, so we save collective time. More importantly, we can retry publishing `react-native-bridge` in isolation which should take 2-3 min, as opposed to the whole flow which is around ~15-18min. (including CircleCI tasks)

Note that it might be easier to review this PR commit by commit.

**To test:**
* Since Buildkite was able to successfully publish the artifacts, we just need to ensure the editor loads without a crash using the [WPAndroid installable build](https://github.com/wordpress-mobile/WordPress-Android/pull/15069#issuecomment-886490063).
